### PR TITLE
Add Unit Tests for InputEventKey

### PR DESCRIPTION
**Pull request by rocketman1243**
This PR adds unit tests for the InputEventKey class.

I tried to test both happy paths and failure paths, such that each part of the if statement gets executed.
If you have any comments or remarks, please let me know!

This PR is part of #43440.

### CODE REVIEW

**Review by Geometror — CHANGES_REQUESTED** [review]
Looks good :)
Just a few style nitpicks: For some variable names you use camelCase, but snake_case should be used instead. (e.g. `noneKey`, `refKey`, `noneRef`,...)

File: tests/core/input/test_input_event_key.h [hunk 1]
```diff
@@ -0,0 +1,296 @@
+/*************************************************************************/
+/*  test_input_event_key.h                                                        */
```
Comment (Geometror) [inline tests/core/input/test_input_event_key.h:2]:
```suggestion
/*  test_input_event_key.h                                               */
```

**Review by Geometror — DISMISSED** [review]

**Review by akien-mga — APPROVED** [review]
Force pushed a rebase to solve style issues, should be good to go now.

I didn't do a thorough review of the actual tests, I think we might want to check a bit more in depth and see if we don't want to further improve the logic of InputEventKey.

For example the fact that physical/non-physical state is based solely on the value of `keycode` being `NONE` or not, and the text representation of this event.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**